pennylane/ops/qutrit/observables.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2018-2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the discrete-variable quantum observables,
+excepting the Pauli gates and Hadamard gate in ``non_parametric_ops.py``.
+"""
+from pennylane import numpy as np
+
+import pennylane as qml  # pylint: disable=unused-import
+from pennylane.ops.qubit import Hermitian
+from pennylane.ops.qutrit import QutritUnitary
+
+
+class THermitian(Hermitian):
+    r"""
+    An arbitrary Hermitian observable for qutrits.
+
+    For a Hermitian matrix :math:`A`, the expectation command returns the value
+
+    .. math::
+        \braket{A} = \braketT{\psi}{\cdots \otimes I\otimes A\otimes I\cdots}{\psi}
+
+    where :math:`A` acts on the requested wires.
+
+    If acting on :math:`N` wires, then the matrix :math:`A` must be of size
+    :math:`3^N\times 3^N`.
+
+    **Details:**
+
+    * Number of wires: Any
+    * Number of parameters: 1
+    * Gradient recipe: None
+
+    Args:
+        A (array): square hermitian matrix
+        wires (Sequence[int] or int): the wire(s) the operation acts on
+        do_queue (bool): Indicates whether the operator should be
+            immediately pushed into the Operator queue (optional)
+        id (str or None): String representing the operation (optional)
+    """
+
+    # This method is overridden to update the docstring.
+    @staticmethod
+    def compute_matrix(A):
+        r"""Representation of the operator as a canonical matrix in the computational basis (static method).
+
+        The canonical matrix is the textbook matrix representation that does not consider wires.
+        Implicitly, this assumes that the wires of the operator correspond to the global wire order.
+
+        .. seealso:: :meth:`~.THermitian.matrix`
+
+        Args:
+            A (tensor_like): hermitian matrix
+
+        Returns:
+            tensor_like: canonical matrix
+
+        **Example**
+
+        >>> A = np.array([[6+0j, 1-2j, 0],[1+2j, -1, 0], [0, 0, 1]])
+        >>> qml.Hermitian.compute_matrix(A)
+        [[ 6.+0.j  1.-2.j  0.+0.j]
+         [ 1.+2.j -1.+0.j  0.+0.j]
+         [ 0.+0.j  0.+0.j  1.+0.j]]
+        """
+        return Hermitian.compute_matrix(A)
+
+    # This overrides `Hermitian.eigendecomposition`, because keying into the dictionary `THermitian_eigs` directly
+    # does not work as expected and users need to key into `Hermitian._eigs` instead without this override.
```

Comment:
I don't quite understand the comment here, what specifically happens when the method is not overridden?

Reply:
Before overriding, if a user tried to key into `THermitian._eigs`, a `KeyError` would be thrown due to the key not being present in the dictionary, but it would work if a user tried keying into `Hermitian._eigs`.

Reply:
> When exactly is `KeyError` raised?

It would get raised when a user tries to access `THermitian._eigs` directly using the flattened Hermitian matrix as a key rather than using the `eigendecomposition` property. This wouldn't be a problem in a real-world scenario as the property is meant to be private, which is why I'm wondering if it's worth just reverting the changes and keying into `Hermitian._eigs` for the test suite.

> Also, if we go with the current solution, let's have a dedicated `_eigs` class attribute for `THermitian` because we're mutating `Hermitian`'s `_eigs` now.

Sure thing. I'll add that attribute to `THermitian` for now, although I'm also wondering whether the change I suggested about just updating the test suite to key into `Hermitian._eigs` rather than `THermitian._eigs` would be better in the long run as it would remove a lot of duplicate code. Let me know what you think! :)

Reply:
> about just updating the test suite to key into Hermitian._eigs rather than THermitian._eigs would be better in the long run as it would remove a lot of duplicate code

This sounds a bit confusing: if I understand correctly, `Hermitian._eigs` would handle both the qubit and the qutrit cases. I think it makes sense to have them separated by having a dedicated `THermitian._eigs` attribute.